Dockerfile

```diff
@@ -1,5 +1,5 @@
-FROM centos:7
-RUN yum install -y python-devel file zip gcc libffi-devel && yum clean all
-RUN curl https://bootstrap.pypa.io/pip/2.7/get-pip.py -o get-pip.py && python get-pip.py && rm get-pip.py
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
+RUN microdnf update -y && microdnf -y install which procps-ng git libffi-devel file xz bzip2 tar zip python3.11 python3.11-pip && microdnf clean all
+RUN python3 -m pip install --upgrade pip
```

Comment:
`--no-cache-dir` is recommended in Dockerfile
```
python3 -m pip install --no-cache-dir --upgrade pip 
```

Reply:
Updated